src/PackageSelection/PackageSelection.php

```diff
@@ -461,4 +491,26 @@ private function filterRepositories(array $repositories)
             return true;
         });
     }
+
+    /**
+     * @param Pool $pool
+     * @param $name
+     * @param Link $link
+     * @return array|\Composer\Package\PackageInterface[]
+     */
+    private function findMatches(Pool $pool, $name, Link $link)
+    {
+        try {
+            if (!$this->bestCandidateStrategy) {
+                return $pool->whatProvides($name, $link->getConstraint(), true);
+            } else {
+                $selector = new VersionSelector($pool);
+                if ($best = $selector->findBestCandidate($name, str_replace(['[', ']'], ['', ''], $link->getConstraint()->getPrettyString()))) {
+                    return [$best];
+                }
+            }
+        } catch (\Exception $e) {
```

Comment:
I'm having some Exception with some libraries that has a wrong version like ">=3.0<4" or "7.0.0-7.0.2" or something like that. Composer doesn't recover from that problem so I have to do it for myseft. I think it's a Composer not Satis problem.